src/PIL/Image.py

```diff
@@ -527,15 +527,19 @@ def _new(self, im):
     def __enter__(self):
         return self
 
+    def _close_fp(self):
+        if getattr(self, "_fp", False):
+            if self._fp != self.fp:
+                self._fp.close()
+            self._fp = DeferredError(ValueError("Operation on closed image"))
+        if self.fp:
+            self.fp.close()
+
     def __exit__(self, *args):
-        if hasattr(self, "fp") and getattr(self, "_exclusive_fp", False):
-            if getattr(self, "_fp", False):
-                if self._fp != self.fp:
-                    self._fp.close()
-                self._fp = DeferredError(ValueError("Operation on closed image"))
-            if self.fp:
-                self.fp.close()
-        self.fp = None
+        if hasattr(self, "fp"):
+            if getattr(self, "_exclusive_fp", False):
```

Comment:
Out of curiosity: does `exclusive_fp` mean, like I think it does, "we own this fp because we opened it"? If so, does it make sense for `.close()` to also close FPs that aren't "owned" by this instance?

Reply:
> Out of curiosity: does exclusive_fp mean, like I think it does, "we own this fp because we opened it"?

Yes

https://github.com/python-pillow/Pillow/blob/697c24bc7dee8a679ae18d6267cc333f79a32dca/src/PIL/Image.py#L3250-L3258

> If so, does it make sense for .close() to also close FPs that aren't "owned" by this instance?

See #5309